### PR TITLE
Added force move undeploy to the construction yards

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/OpenRA/ra2/issues/131.

### CODE REVIEW

File: mods/ra2/rules/allied-structures.yaml [hunk 1]
```diff
@@ -30,6 +30,21 @@ gacnst:
...
 		DeployCursor: undeploy
 		RequiresCondition: factundeploy && !build-incomplete
+	TransformsIntoMobile:
+		RequiresCondition: factundeploy
+		Locomotor: tracked
+		Voice: Move
+		RequiresForceMove: true
+	TransformsIntoRepairable:
+		RequiresCondition: factundeploy
+		RepairActors: gadept
```
Comment (MustaphaTR) [inline mods/ra2/rules/allied-structures.yaml]: These should list all 3 (caoutp too). Faction's one.

File: mods/ra2/rules/soviet-structures.yaml [hunk 2]
```diff
@@ -32,6 +32,21 @@ nacnst:
...
 		DeployCursor: undeploy
 		RequiresCondition: factundeploy && !build-incomplete
+	TransformsIntoMobile:
+		RequiresCondition: factundeploy
+		Locomotor: tracked
+		Voice: Move
+		RequiresForceMove: true
+	TransformsIntoRepairable:
+		RequiresCondition: factundeploy
+		RepairActors: nadept
```
Comment (MustaphaTR) [inline mods/ra2/rules/soviet-structures.yaml]: Same here.

File: mods/ra2/rules/soviet-structures.yaml [hunk 3]
```diff
@@ -32,6 +32,21 @@ nacnst:
...
 		DeployCursor: undeploy
 		RequiresCondition: factundeploy && !build-incomplete
+	TransformsIntoMobile:
+		RequiresCondition: factundeploy
+		Locomotor: tracked
+		Voice: Move
+		RequiresForceMove: true
+	TransformsIntoRepairable:
+		RequiresCondition: factundeploy
+		RepairActors: nadept, caoutp, gadept
```
Comment (MustaphaTR) [inline mods/ra2/rules/soviet-structures.yaml]: `gadept, nadept, caoutp` please. Same applies allied one.

Comment (Mailaender) [reply mods/ra2/rules/soviet-structures.yaml]: It is exactly the same just in a different order.

Comment (MustaphaTR) [reply mods/ra2/rules/soviet-structures.yaml]: Yes, for consistency. I'm sure other places (Repairable in defaults) use that order.

File: mods/ra2/rules/soviet-structures.yaml [hunk 4]
```diff
@@ -32,6 +32,21 @@ nacnst:
 		Facing: 96
 		DeployCursor: undeploy
 		RequiresCondition: factundeploy && !build-incomplete
+	TransformsIntoMobile:
+		RequiresCondition: factundeploy
+		Locomotor: tracked
```
Comment (MustaphaTR) [inline mods/ra2/rules/soviet-structures.yaml]: MCVs have `heavytracked` locomotor.

File: mods/ra2/rules/soviet-structures.yaml [hunk 5]
```diff
@@ -32,6 +32,21 @@ nacnst:
...
+		Voice: Move
+		RequiresForceMove: true
+	TransformsIntoRepairable:
+		RequiresCondition: factundeploy
+		RepairActors: nadept, gadept, caoutp
```
Comment (MustaphaTR) [inline mods/ra2/rules/soviet-structures.yaml]: This is till wrong.

Comment (Mailaender) [reply mods/ra2/rules/soviet-structures.yaml]: `nadept` is the soviet counter-part. Why is this wrong for the soviet MCV?

Comment (MustaphaTR) [reply mods/ra2/rules/soviet-structures.yaml]: All units can get repaired from all Service depots, i don't see any reason why Soviet Service Depot should be first for Soviet MCV.

Comment (Mailaender) [reply mods/ra2/rules/soviet-structures.yaml]: Because that is most likely in a safe place at the mother base.

Comment (MustaphaTR) [reply mods/ra2/rules/soviet-structures.yaml]: I think changing the order would be an unnecessary inconsistency in the code. The order changes nothing ingame.

Comment (Mailaender) [reply mods/ra2/rules/soviet-structures.yaml]: Why are you making me change this all the time if it doesn't do anything? I think my way of setup is more consistent with the faction specific buildings.

**Pull request merged**